spec/integration/solved_spec.rb

```diff
@@ -366,4 +366,61 @@
       expect(response.status).to eq(200)
     end
   end
+
+  context "with discourse-assign installed", if: defined?(DiscourseAssign) do
+    let(:admin) { Fabricate(:admin) }
+    fab!(:group)
+
+    before do
+      SiteSetting.solved_enabled = true
+      SiteSetting.assign_enabled = true
+      SiteSetting.enable_assign_status = true
+      SiteSetting.assign_allowed_on_groups = "#{group.id}"
+      SiteSetting.assigns_public = true
+      SiteSetting.assignment_status_on_solve = "Done"
+
+      group.add(p1.acting_user)
+      group.add(user)
+
+      sign_in(user)
+    end
+
+    it "update all assignments to this status when a post is accepted" do
+      assigner = Assigner.new(p1.topic, user)
+      result = assigner.assign(user)
+      expect(result[:success]).to eq(true)
+
+      expect(p1.topic.assignment.status).to eq("New")
+      DiscourseSolved.accept_answer!(p1, user)
+
+      expect(p1.reload.custom_fields["is_accepted_answer"]).to eq("true")
+      expect(p1.topic.assignment.reload.status).to eq("Done")
+    end
+  end
+
+  context "with using assigns_reminder_assigned_topics_query modifier" do
```

Comment:
I don't know on this looking at it again. Since the test above relies directly on the discourse-assign plugin, maybe calling the AssignReminder class directly for testing is best? @jjaffeux want to weigh in here as well?